### ISSUE TITLE
A badge for UI translators

### ISSUE THREAD

**Issue by RyckRichards** [Tatoeba/tatoeba2#2481]
Some people haven't joined us on Transifex yet or stop contributing for one (or more) of the following reasons:

- They weren't aware of that. I've been inviting them but there are so many to be invited yet;
- Some notice that their job gets unnoticed. Nobody knows who translate the UI into that language and how much he/she contributed (even the translator itself);
- As the way it is now, it seems that this task isn't important enough.

Recently, po_slovensky translated the entire UI into Slovak and MarijnKp for Gronings. Few people will know that (just some who read the post on the Wall and people who are in charge of the maintenance of the UI translation).

That said, I'd suggest adding a badge on the translator's profile on Tatoeba. (A golden star, maybe?)* It'll make people realize that this task exist, motivate them to translate/proofread they UI and also people would be able to know who they can report mistakes/suggest improvements. Also, it would express a kind of gratitude (We as Tatoeba Organization recognize this translator's efforts on translating the Tatoeba UI)

* A star with the saying "This translator is contributing to the translation of the UI" and maybe a button like "would you like to help us?

**Comment by soliloquist-tatoeba** [comment]
> It'll make people realize that this task exist, motivate them to translate/proofread they UI and also people would be able to know who they can report mistakes/suggest improvements.

The UI translators can give information about it on their profiles too, but I liked the idea of badges. 

Some web sites place multiple badges/medals on the profiles or under the avatars of their members to motivate and reward them. The same can be done on Tatoeba to increase motivation and persistence (especially of the younger members). Not only for translating the UI or Wiki, but also for other categories like the amount of sentences, comments, tags, reviews and audio they have contributed.

The levels can be like the following:

A clover for 100-999 contributions

A bronze medal for 1,000-9,999 contributions 

A silver medal for 10,000-24,999 contributions

A gold medal for 25,000-49,999 contributions

A sapphire for 50,000-99,999 contributions

An emerald for 100,000-149,999 contributions

A ruby for 150,000-199,999 contributions

A diamond for 200,000+ contributions

(The ranges may change from category to category)

Napoleon was aware of how useful they were long time ago...

![1](https://user-images.githubusercontent.com/47579687/89459122-d8184680-d770-11ea-9116-bcc74659b26e.jpg)
![2](https://user-images.githubusercontent.com/47579687/89459136-db133700-d770-11ea-9068-6b8881b3d91d.jpg)

**Comment by jiru** [comment]
@soliloquist-tatoeba While I appreciate your idea, I wonder if the sole raw number of contributions should be the only incentive. It might as well favour quantity over quality. What about contributors who spend a lot of time to come up with accurate and natural-sounding translations, or realistic contexts for original sentences?

Some more food for thoughts: https://en.wikipedia.org/wiki/Gamification

**Comment by soliloquist-tatoeba** [comment]
Thanks for your comment and for the link.

I understand your concern about the balance between quality and quantity. I think introducing some non-quantity-based badges may help users to focus on quality and prevent them from being completely number-driven. For example, there could be a light bulb badge for the users who regularly create original and interesting, non-monotonous content. Similarly, there could be a green shield with a check mark for the users who are known to contribute accurate and natural-sounding sentences/translations. Since they are not based on quantity, one member may have contributed 50,000 sentences but still doesn't have them, while another member with 5,000 contributions may have both of them.

The quantity-based badges could be granted automatically. But naturally, it would be more difficult to determine who deserves the non-quantity-based ones. My opinion is that such badges should be granted by an admin, with the reference/approval of the corpus maintainers of the regarding language.

This issue was created for UI translations, but based on the same feature, many other badge types could be created over time to stimulate motivation, quality and creativity, once it is implemented.

**Comment by RyckRichards** [comment]
I believe that a "badge" can be added manually by an admin and/or corpus maintainers. People will make sure that content from people who own a badge would be good.

As for translations, I'm checking regularly how much people are getting involved.

**Comment by jiru** [comment]
@soliloquist-tatoeba As you pointed out we are getting out a little bit of the scope of the initial issue, but it’s interesting. As far as I remember, the topic of gamification has been brought up by liori at some point and it was also a [2015 GSOC idea](https://en.wiki.tatoeba.org/articles/show/gsoc_2015_ideas#achievement-system). Feel free to open a new issue to elaborate.

I think what you @RyckRichards mentioned is a good way to do it (have an admin/CM manually give out badges).

**Comment by alanfgh** [comment]
I am uncomfortable with the idea of badges altogether, but **especially** when it comes to having them granted on a subjective basis by admins/CMs. I don't want to take time away from my meaningful tasks (flagging and fixing errors, contributing my own sentences, answering questions, reporting issues) to decide who should get a tiny colored icon. I don't want people coming to me to ask for badges for themselves or others. I don't want to have to resolve arguments about people who supposedly received badges unfairly. I don't want to feel as though I'm falling behind in an arms race because other admins/CMs are giving out badges for certain kinds of contributions that I don't think are worthwhile.

That's the load on admins/CMs. Then there's the load on developers. I would hate to see people spending their time on this rather than fixing bugs that pertain to actual core functionality -- and we have a big backlog already. Remember that it's not just a question of design and implementation. This will invariably involve introduction of bugs in their own right.

And then there's the effect on the atmosphere for all contributors. One argument over someone unfairly getting a badge has the potential to undo anything positive we could get from such a system. Instead of people quitting because NO ONE is getting a badge, we'll get people quitting because SOMEONE ELSE is getting a badge. 

At Stack Exchange, which has an intricate badge system, I find that badges have zero effect on my desire to contribute. 

People should be motivated by pride in the work they do, in terms of both quantity and quality. Almost every member of this community has made contributions that can't be seen from their profile, and that's the way it should be. Trying to develop a system in which every contribution is rewarded publicly will fail, and along the way, it will introduce all kinds of negative unintended consequences.

**Comment by soliloquist-tatoeba** [comment]
I agree that the subjectivity of giving some badges may cause issues, and I'm in favor of a system that is as objective (and even algorithmic) as possible, but I don't think subjectivity is something we can completely avoid. I have seen multiple times native speakers arguing over a link, tag or review mark on a sentence. The current tools have that potential, too. Some may find a reward system useless, but others may be motivated by it.

If one admin doesn't want to spend time for such a task, I think they can simply say so, and they won't be contacted for it.

I wonder what the other admins and especially @ckjpn think about this idea. He regularly works on data exports and shares lists, rankings and stats about the contributions and members voluntarily. Maybe he doesn't mind dealing with it.

**Comment by alanfgh** [comment]
> I agree that the subjectivity of giving some badges may cause issues, and I'm in favor of a system that is as objective (and even algorithmic) as possible

**If** we were going to have a badge system, I would **only** want badges that are objective and, preferably algorithmic.

> but I don't think subjectivity is something we can completely avoid

Yes, we can -- by not granting badges other than those that have an objective basis.

> I have seen multiple times native speakers arguing over a link, tag or review mark on a sentence.

Links and tags (and, to some extent, review marks) are something we need. The fact that subjectivity is required to apply them is outweighed by their utility. Badges are something we've been fine without.

> ... regularly works on data exports and shares lists, rankings and stats about the contributions and members voluntarily. Maybe he doesn't mind dealing with it.

You're missing my point. I'm not claiming that we couldn't find people who would be willing to hand out honorary sapphires. I'm claiming that there's no way that it can be done fairly. Quite aside from the fact that the system would take on the biases of the admins/CMs who have the time and inclination to do this **within a specific language**, we have no admins or CMs who speak the vast majority of the languages on Tatoeba. Thus, no one will be in a position to decide whether contributor X's sentences in Plains Cree are particularly creative and deserve a medal.

I don't deny that getting a badge could motivate some people to some extent. But the process of building and maintaining even a purely algorithmic system is high. Furthermore, even without badges, and in the presence of solely quantitative criteria, we've already seen distortions introduced by gamification, such as large quantities of low-quality sentences. The types of distortions that would be introduced to win a badge for creativity would be different but no less problematic.

**Comment by soliloquist-tatoeba** [comment]
Anyway, if Ricardo's initial request of badges for UI translators doesn't get implemented, adding a new block on the 'Contact us' page, showing the contact information for the UI translators (either as their Tatoeba profile pages or as email addresses) can be considered as an alternative solution.

https://tatoeba.org/eng/contact

Then that part can be localized to different languages on Transifex.

**Comment by trang** [comment]
I'm not sure about having the list of UI translators on the "Contact us" page. It seems a bit irrelevant to have them listed there because people will usually not seek to contact UI translators. Or perhaps I don't understand the idea...

Having "Credits" page or a specific page that lists all the UI translators would make more sense.

**Comment by soliloquist-tatoeba** [comment]
> It seems a bit irrelevant to have them listed there because people will usually not seek to contact UI translators. Or perhaps I don't understand the idea...

This is from Ricardo's first message. 

"and also people would be able to know who they can report mistakes/suggest improvements."

I thought one way to provide it was to show the contact information of translators on the 'Contact us' page.

I meant something like this.

"The UI is translated into X by Y and Z. If you find any errors or want to suggest any improvements, you could contact them in X."

> Having "Credits" page or a specific page that lists all the UI translators would make more sense.

Yes, that would work as well, but I think it would still be good to have a link of that page on the 'Contact us' page, too. People usually look for a contact page first to report errors.

**Comment by jiru** [comment]
@alanfgh You are right to point out the potential negative impacts of rewards. The question of whether contributors should be kept motivated by rewards, or solely by the pride of their own work, is quite philosophical. I don’t think there is a right or wrong answer to that, it’s just a (political) decision to make. A user [I interviewed](https://en.wiki.tatoeba.org/articles/show/ux-test-3#debrief) pointed out Tatoeba is boring because of the lack of feedback from contributions.

That said, my previous message, though vague, was solely about UI translations (the scope of this ticket). And I didn’t mean that corpus maintainers should be involved into judging the quality of UI translations. The badge handling process I had in mind was the following:

1. Receive a badge request from a UI translators as a PM.
1. Somehow manually figure out that username UTa on Tatoeba is the same person as username UTr on Transifex.
2. Check if UTr has contributed at least one string on Transifex.
3. If yes, give out the badge to UTa on Tatoeba.

This process could be fully automated by some OAuth magic or something, but such automation would be likely costly. That’s why I suggested manually giving out badges as a first step.

**Comment by soliloquist-tatoeba** [comment]
> A user I interviewed pointed out Tatoeba is boring because of the lack of feedback from contributions.

I agree with her.

I have seen a lot of new users adding translations vigorously for several days or sessions, and then leaving for good. My guess is that, at some point, they find the general atmosphere monotonous and lose their motivation. There is no reward system or sense of achievement. If we want to get more contributions from the new generation, I think this should be something to think over.

See how Duolingo does it.

https://duolingo.fandom.com/wiki/Achievements

https://www.uisources.com/explainer/duolingo-gamification-badges

Even the streak achievement (the wildfire badge) alone may have an important role on persistence. Imagine that there is a similar system on Tatoeba. Many young/new users would try to login every day and do some contributions to get the badge or not to lose the one they have. And after a while, Tatoeba would become a habit for them.

**Comment by RyckRichards** [comment]
Thanks for your comments, everyone!

I've read some relevant comments and I'd like to point out some stuff.

> My guess is that, at some point, they find the general atmosphere monotonous and lose their motivation. There is no reward system or sense of achievement.

That "system" I'm suggesting won't solve the problem for sure but it would help. For sentences, it's easy to check how much you've contributed since there are "numbers" on your profile and your username on your sentences. That doesn't happen to UI translations. People leave Tatoeba for numerous reasons, but we'd *avoid* them stoping translating the UI because we didn't do anything about it. And even they leave, everybody would be able to know that this user helped Tatoeba and language learners / native speakers of that language. No system is "fail-proofed" but I believe we can try something.

> This process could be fully automated by some OAuth magic or something, but such automation would be likely costly. That’s why I suggested manually giving out badges as a first step.

I've suggested a kind of "manual reward" (or "I confirm you've helped us by translating the UI") because

1 - We would make it more "human" since people would "confirm" that people helped Tatoeba;
2 - It would require less effort from the dev team
3 - We could make sure to give the translator a badge if they helped somehow 

> I'm not claiming that we couldn't find people who would be willing to hand out honorary sapphires. I'm claiming that there's no way that it can be done fairly. 

People who are willing to do so can have a look on Transifex - https://www.transifex.com/tatoeba/tatoeba_website/dashboard/ - and have some clue. Also, they can ask me, too since I've been trying to organize and find people to translate the UI. Again, no system would be 100% fair but if we go this way, we'd do anything. 
Take for example the "promotion" for AC and CM. Some of them left, some of them stopped contributing, I myself committed some mistakes... Were all the "promotions" fair or a legitim "We (Tatoeba community) trust you since we've seen how much you've contributed, what you're willing to do, etc? Or both (fair and legitims)? I'd say that they were fair AND legitims

> Quite aside from the fact that the system would take on the biases of the admins/CMs who have the time and inclination to do this within a specific language, we have no admins or CMs who speak the vast majority of the languages on Tatoeba. Thus, no one will be in a position to decide whether contributor X's sentences in Plains Cree are particularly creative and deserve a medal.

It would be great if we had more people engaged on Tatoeba who can speak more languages but we can work with we get - which as much. There are statistics on Transifex, people who knows which users contributed and so on. Also, it's a "thank you" and "This user helped us" message, not any kind of "Hey, this user is the best one of Tatoeba because....". It's a system which would value efforts. 
As from the last months, we've been able to add Serbian, Gronings, Slovak and other languages on the UI. I just wonder how many more we could have if we just have a "Yes, this user helped" system.

> If one admin doesn't want to spend time for such a task, I think they can simply say so, and they won't be contacted for it.
> I don't want to take time away from my meaningful tasks (flagging and fixing errors, contributing my own sentences, answering questions, reporting issues) to decide who should get a tiny colored icon

Yes! I just mentioned admins and CMs since they are the ones who have been working on Tatoeba more time than other users and know Tatoeba a lot. But if an admin or a CM don't want to do so, it's ok too. It'd be a volunteer job (who wants to have this extra task?)
Also, it's a "thank you" and it can mean a lot to many people, I believe. That's why I suggested this. 

> I would hate to see people spending their time on this rather than fixing bugs that pertain to actual core functionality -- and we have a big backlog already. 

There are not so many people evolved as for now and again, it's an extra task which people can assign themselves to do so or not. Nothing on Tatoeba is mandatory. :) We do what we do because we love Tatoeba and languages.

**Comment by jiru** [comment]
What if you Ricardo were in charge or handling badges (or whatever kind of recognition displayed on the profile page) to UI translators? After all, you must be most knowledgeable about who’s working on UI translations now. :smile:

**Comment by RyckRichards** [comment]
I'd be glad to do such thing :)
